### ISSUE TITLE
Error detecting git branch: expected branch name starting with refs/heads/, got: refs/tags/v...

### ISSUE THREAD

**Issue by laurrentt** [launchdarkly/ld-find-code-refs#155]
Using the provided Github Action workflow, when a tag is created, the find-code-ref action is triggered but fails because it doesn't seem to support tags refs.

**Comment by vlymar** [comment]
Thanks for the bug report! We'll take a look soon.

**Comment by Andrei-Predoiu** [comment]
We are experiencing exactly the same issue in CircleCI
It makes it impossible to do release based deployments using the orb

**Comment by arun-selva** [comment]
We couldnt deploy release tags with launchdarkly and will be great if this feature is made available. Thanks

**Comment by InTheCloudDan** [comment]
@Andrei-Predoiu the source of the error @laurrentt mentioned is coming from our Github Actions implementation. Can you confirm the exact error message you are receiving when trying to use this with CircleCI?

**Comment by Andrei-Predoiu** [comment]
I cannot say if the errors stem from the same root cause but here is an example of a CircleCi failed job that is trying to run on a Release tag:

## Checkout code:
```
Using SSH Config Dir '/root/.ssh'
git version 2.18.1
Cloning git repository
Cloning into '.'...
Warning: Permanently added the RSA host key for IP address '140.82.112.3' to the list of known hosts.
remote: Enumerating objects: 5425, done.        
remote: Counting objects: 100% (1643/1643), done.        
remote: Compressing objects: 100% (574/574), done.        
remote: Total 5425 (delta 774), reused 1379 (delta 621), pack-reused 3782        
Receiving objects: 100% (5425/5425), 663.59 KiB | 10.37 MiB/s, done.
Resolving deltas: 100% (2651/2651), done.
Checking out tag
Note: checking out 'release-VP-0.1'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at a5998a2 VP-1527 replaced order, trying to fix bad env
HEAD is now at a5998a2 VP-1527 replaced order, trying to fix bad env
```

## Find flag references:
```
#!/bin/sh -eo pipefail
ld-find-code-refs \
  -debug=true \
  -accessToken=api-1ad454cb-7fc4-4123-b9b5-ed5a24954310 \
  -projKey=default \
  -exclude= \
  -contextLines=2 \
  -baseUri=https://app.launchdarkly.com \
  -repoType=custom \
  -repoUrl= \
  -defaultBranch=master \
  -delimiters= \
  -commitUrlTemplate= \
  -hunkUrlTemplate= \
  -repoName=${CIRCLE_PROJECT_REPONAME} \
  -updateSequenceId=${CIRCLE_BUILD_NUM} \
  -dir=/repo

ld-find-code-refs version 1.3.0
DEBUG: 2021/07/14 13:24:37 git.go:67: identified branch name: HEAD
ERROR: 2021/07/14 13:24:37 coderefs.go:64: error parsing git branch name: git repo at /repo must be checked out to a valid branch or --branch option must be set

Exited with code exit status 1
```

**Comment by InTheCloudDan** [comment]
@Andrei-Predoiu understood, thanks for providing the logs it will help us narrow it down.

**Comment by laupow** [comment]
@carmenquan @jazanne what's the timeline for the next release? 

Looking forward to the 2.3.x release for use with Circle CI ([orb link](https://circleci.com/developer/orbs/orb/launchdarkly/ld-find-code-refs)) 🙏

**Comment by carmenquan** [comment]
@laupow We are preparing a release, but don't have a timeline just yet. I'll update this thread once something changes. Thanks for your patience!

**Comment by carmenquan** [comment]
Thanks for everyone's patience, we've released v2.4.0, which includes the fix to this issue 🎉 

https://github.com/launchdarkly/ld-find-code-refs/releases/tag/2.4.0
https://circleci.com/developer/orbs/orb/launchdarkly/ld-find-code-refs

**Comment by adomass** [comment]
Now we get error like this
ERROR: 2021/12/08 13:55:49 github-actions.go:49: error detecting git branch: expected ref name starting with refs/heads/ or refs/tags/, got: refs/tags/2021.12.5.1375
